### ISSUE TITLE
OS4 migration of exported meeting-data files

### ISSUE THREAD

**Issue by r-peschke** [OpenSlides/openslides-backend#1463]
An older, perhaps OS3 meeting-export will pass a migration process, when being imported into OS4. This in-memory migration consumes a lot of memory and hinder the import.
We could separate the migration, always or selectable, from import.

**Comment by jsangmeister** [comment]
We cannot, because without the migration the data will be invalid (e.g. fields will exist which do not exist in our code base anymore and vice-versa). How would you imagine this to work?

**Comment by r-peschke** [comment]
> We cannot, because without the migration the data will be invalid (e.g. fields will exist which do not exist in our code base anymore and vice-versa). How would you imagine this to work?

I changed the description of the idea to make it more clearly

**Comment by jsangmeister** [comment]
Ok, that sounds like a plan. But does this really help that much? Total time consumed is still the same (or even higher, because of manual intervention) and you still have to do the whole act on the server, so you cannot even do it offline (except if you run the backend locally, which I don't think we want to offer to users as a real approach).

**Comment by r-peschke** [comment]
> Ok, that sounds like a plan. But does this really help that much? Total time consumed is still the same (or even higher, because of manual intervention) and you still have to do the whole act on the server, so you cannot even do it offline (except if you run the backend locally, which I don't think we want to offer to users as a real approach).

I don't think that it is a good idea, but I have a problem importing a 80MB Json-meeting-file from migration level 3 on a 16GB-memory hardware.The migration needs nearly 8 Minutes and after migration the memory is nearly exhausted (seen with htop).
In such a situation this would help. We have to watch this on a production setup, but 80MB is not big and we will have more migrations and next year maybe a lot of hosted customers, that want to migrate their OS3-meetings, occasionally in parallel.

**Comment by jsangmeister** [comment]
Maybe we should then into the migration code and optimize that instead of implementing a workaround which does not even help that much. It sounds like the in-memory migration has memory leaks and pollutes the RAM with that. Maybe you can take a look at that?

**Comment by r-peschke** [comment]
> Maybe we should then into the migration code and optimize that instead of implementing a workaround which does not even help that much. It sounds like the in-memory migration has memory leaks and pollutes the RAM with that. Maybe you can take a look at that?

The in-memory-migration uses the standard-migration-code, the only exception: all migration-stages, that are stored in the special migration tables, are stored in memory. After migration/import finished/aborted the backend-server continues to run with usual consume of memory.

**Comment by jsangmeister** [comment]
Then the normal migration is maybe coded inefficiently. An 80mb file should not produce 16gb worth of data...
